bentoml/_internal/bento/build_config.py

```diff
@@ -2,41 +2,50 @@
 import re
 import typing as t
 import logging
+from abc import abstractmethod
 from sys import version_info as pyver
+from typing import TYPE_CHECKING
 
 import fs
 import attr
 import yaml
 import fs.copy
-from fs.base import FS
-from piptools.scripts.compile import cli as pip_compile_cli  # type: ignore
+from piptools.scripts.compile import cli as pip_compile_cli
 
+from .gen import generate_dockerfile
 from ..utils import bentoml_cattr
 from ..utils import resolve_user_filepath
 from ..utils import copy_file_to_fs_folder
-from .docker import ImageProvider
+from .docker import DistroSpec
+from .docker import DEFAULT_CUDA_VERSION
+from .docker import CUDA_SUPPORTED_DISTRO
+from .docker import DEFAULT_DOCKER_DISTRO
+from .docker import CONDA_SUPPORTED_DISTRO
+from .docker import SUPPORTED_CUDA_VERSION
+from .docker import DOCKER_SUPPORTED_DISTRO
+from .docker import SUPPORTED_PYTHON_VERSION
```

Comment:
Hm, doesn't it make more sense for these to be here and for docker to get them from the bento yaml?

Reply:
sure

Reply:
These are fields for validation. But I don't have a strong opinion on this. I think it is just nicer to leave those under docker since I defined a lot of variables and spec there.